packages/faucet/src/api/webserver.ts

```diff
@@ -58,21 +59,32 @@ export class Webserver {
           // context.request.body is set by the bodyParser() plugin
           const requestBody = context.request.body;
           const creditBody = RequestParser.parseCreditBody(requestBody);
-
           const { address, denom } = creditBody;
 
           if (!isValidAddress(address, constants.addressPrefix)) {
             throw new HttpError(400, "Address is not in the expected format for this chain.");
           }
 
-          const availableTokens = await faucet.availableTokens();
+          const entry: Date | undefined = this.addressCounter.get(address);
+          if (entry !== undefined) {
+            if (entry.getDate() + 24 * 3600 > Date.now()) {
+              throw new HttpError(
+                405,
+                "Too many request from the same address. Blocked to prevent draining. Please wait 24h and try it again!",
+              );
+            }
+          }
+
+          const availableTokens: string[] = await faucet.availableTokens();
```

Comment:
No need to change this line as part of this PR, right?

```suggestion
          const availableTokens = await faucet.availableTokens();
```

Reply:
Was my IDE somehow... Gonna delete it